deploy/swrs/public/view/carbon_tax_calculation.sql

```diff
@@ -8,86 +8,109 @@ begin;
+      fd.fuel_type,
+      fd.emission_category,
+      fd.fuel_amount,
+      _fuel_charge.fuel_charge,
+      round((fuel_amount * _fuel_charge.fuel_charge), 2) as calculated_carbon_tax from fuel_data fd
+    join swrs.report r
+        on r.id = fd.report_id
+    join swrs.facility as _facility
+        on r.id = _facility.report_id
+        and _facility.facility_type != 'LFO'
+    join swrs.carbon_tax_act_fuel_type as _cta
+      on carbon_tax_act_fuel_type_id = _cta.id
+    join swrs.fuel_charge as _fuel_charge
+      on _fuel_charge.carbon_tax_act_fuel_type_id = _cta.id
+      and (concat(r.reporting_period_duration::text, '-12-31')::date
+      between _fuel_charge.start_date and _fuel_charge.end_date);
+
+comment on view swrs.carbon_tax_calculation is 'Calculates carbon tax for each fuel type based on the data in the Schedule 2 table found in the carbon tax act https://www.bclaws.gov.bc.ca/civix/document/id/complete/statreg/08040_01';
+comment on column swrs.carbon_tax_calculation.report_id is 'The id of the report each fuel was reported for';
+comment on column swrs.carbon_tax_calculation.fuel_type is 'The fuel type reported';
+comment on column swrs.carbon_tax_calculation.emission_category is 'The emission category the reported fuels falls under. Some categories are taxed and others are not (a carbon_taxed boolean is defined in the swrs.emission_category table). Emission categories are defined in the Greenhouse Gas Industrial Reporting and Control Act here: https://www.bclaws.gov.bc.ca/civix/document/id/lc/statreg/249_2015#section14';
+comment on column swrs.carbon_tax_calculation.fuel_mapping_id is 'The id to the the fuel mapping table matching the fuel type';
+comment on column swrs.carbon_tax_calculation.fuel_carbon_tax_details_id is 'The id to the the fuel carbon tax details table matching the fuel type';
+comment on column swrs.carbon_tax_calculation.carbon_tax_act_fuel_type_id is 'The id to the the carbon tax act fuel type table matching the fuel type, matches the Schedule 2 table in the carbon tax act https://www.bclaws.gov.bc.ca/civix/document/id/complete/statreg/08040_01';
+comment on column swrs.carbon_tax_calculation.fuel_amount is $$
+  The amount of fuel reported for the fuel type.
+  This fuel_amount is the annual_fuel_amount as reported in the swrs report multiplied by the unit_conversion_factor in the fuel_carbon_tax_details table to normalize the units with the rate defined in the carbon tax.
+  Vented and Flared emissions return a fuel_amount that has been converted to a natural gas fuel_amount of m3.
+  X tonnes of Flared CO2 emissions return a record with a Natural gas fuel amount of Y (as per WCI.20 EF for non-marketable gas, Table 20-3: 2.151 kgCO2/m3 non-marketable NG in BC)
+$$;
+comment on column swrs.carbon_tax_calculation.facility_id is 'The id of the facility the fuel was reported for';
+comment on column swrs.carbon_tax_calculation.reporting_year is 'The reporting year the fuel was reported';
```

Comment:
is the reporting year needed here? It can be retrieved through the `report_id` foreign key (if metabase metadata is configured properly).